cmd/process.go

```diff
@@ -170,40 +170,35 @@ func StartProcessor(
 			// if the message is burned or pending, check for an attestation
 			if msg.Status == types.Created || msg.Status == types.Pending {
 				response := circle.CheckAttestation(cfg.Circle.AttestationBaseUrl, logger, msg.IrisLookupId, msg.SourceTxHash, msg.SourceDomain, msg.DestDomain)
-				if response != nil {
-					if msg.Status == types.Created && response.Status == "pending_confirmations" {
-						logger.Debug("Attestation is created but still pending confirmations for 0x" + msg.IrisLookupId + ".  Retrying...")
-						State.Mu.Lock()
-						msg.Status = types.Pending
-						msg.Updated = time.Now()
-						State.Mu.Unlock()
-						time.Sleep(10 * time.Second)
-						requeue = true
-						continue
-					} else if response.Status == "pending_confirmations" {
-						logger.Debug("Attestation is still pending for 0x" + msg.IrisLookupId + ".  Retrying...")
-						time.Sleep(10 * time.Second)
-						requeue = true
-						continue
-					} else if response.Status == "complete" {
-						logger.Debug("Attestation is complete for 0x" + msg.IrisLookupId + ".  Retrying...")
-						State.Mu.Lock()
-						msg.Status = types.Attested
-						msg.Attestation = response.Attestation
-						msg.Updated = time.Now()
-						broadcastMsgs[msg.DestDomain] = append(broadcastMsgs[msg.DestDomain], msg)
-						State.Mu.Unlock()
-					}
-				} else {
-					// add attestation retry intervals per domain here
+
+				if response == nil {
 					logger.Debug("Attestation is still processing for 0x" + msg.IrisLookupId + ".  Retrying...")
-					time.Sleep(10 * time.Second)
-					// retry
 					requeue = true
 					continue
+				} else if msg.Status == types.Created && response.Status == "pending_confirmations" {
+					logger.Debug("Attestation is created but still pending confirmations for 0x" + msg.IrisLookupId + ".  Retrying...")
+					State.Mu.Lock()
+					msg.Status = types.Pending
+					msg.Updated = time.Now()
+					State.Mu.Unlock()
+					requeue = true
+					continue
+				} else if response.Status == "pending_confirmations" {
+					logger.Debug("Attestation is still pending for 0x" + msg.IrisLookupId + ".  Retrying...")
+					requeue = true
+					continue
+				} else if response.Status == "complete" {
+					logger.Debug("Attestation is complete for 0x" + msg.IrisLookupId + ".")
+					State.Mu.Lock()
+					msg.Status = types.Attested
+					msg.Attestation = response.Attestation
+					msg.Updated = time.Now()
+					broadcastMsgs[msg.DestDomain] = append(broadcastMsgs[msg.DestDomain], msg)
+					State.Mu.Unlock()
 				}
```

Comment:
no else case anymore? maybe one for safety / debugging?
(are the only response.Status types `pending_confirmations` and `complete`? )

Reply:
I basically inverted the if statement that was there previously. That removes the need to nest the other if statements and completely removes the need for an else case.

Yes that is correct, only those two options:
![image](https://github.com/strangelove-ventures/noble-cctp-relayer/assets/29391744/18400fb4-4f9e-4380-9d9a-04b1fcdffc2b)